packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -30,26 +30,93 @@ export interface FluidAppInsightsLoggerConfig {
+		 * In order for the filters to be valid they must meet the following conditions:
+		 * 1. There must not be any two filters with the same `namespacePattern`.
+		 * 2. All {@link NamespaceFilter} must not have any defined `namespacePatternException` that is not a child of the parent `namespacePattern`
 		 */
 		filters?: TelemetryFilter[];
 	};
 }
 
 /**
- * Object used with an {@link FluidAppInsightsLoggerConfig}
- * to define logic for filtering of telemetry events
+ * A filter used to match against the category of a telemetry event
  *
  * @public
  */
-export interface TelemetryFilter {
+export interface CategoryFilter {
+	/**
+	 * The categories of telemetry events that this filter applies to
+	 */
+	categories: TelemetryEventCategory[];
+}
+
+/**
+ * A filter used to match against the namespaces of a telemetry event
+ *
+ * @public
+ */
+export interface NamespaceFilter {
+	/**
+	 * The namespace pattern to filter telemetry events.
+	 *
+	 * @remarks This will match namespaces that start with the given string. It is not a Regex pattern.
+	 * @example
+	 * "perf:latency" will match any namespace starting with "perf:latency"
+	 */
+	namespacePattern: string;
 	/**
-	 * The category of telemetry event that this filter applies to
+	 * A list of namespace patterns to explicitly exclude from the filter.
+	 *
+	 * @example If you have a namespacePattern of "perf:latency" but want to exclude
+	 *
+	 * events from "perf:latency:ops", you would add "perf:latency:ops" to this list.
 	 */
-	category: TelemetryEventCategory;
+	namespacePatternExceptions?: Set<string>;
 }
 
+/**
+ * Object used with an {@link FluidAppInsightsLoggerConfig}
+ * to define a filter with logic for matching it to telemetry events.
+ * Filters can include either a category, namespace or both types of filters; a valid filter must have at least one defined.
+ * Not definining the `categories` filter array is the same as providing an array with all possible categories.
+ *
+ * Events must satisify the following rules for a telemetry filter:
+ *
+ * 1. The event must match the requirements of the most specific relevant filter to it. This takes precedence over a more generic filter.
+ * The less categories and longer the namespace within a filter, the more specific it is. Definining no categories is equivalant to defining all categories.
+ *
+ * 2. If a {@link TelemetryFilter} specifies both `categories` and a `namespace`, the event must match both.
+ *
+ * 3. If only `categories` or a `namespace` is provided, the event should just match the with whatever was defined.
+ *
+ * 4. If a `namespace` pattern exception is specified in the {@link TelemetryFilter}, the event should not match the exception pattern.
+ *
+ * @public
+ *
+ * @example With the following configuration, an event `{ namespace: "A.B.C", categories: ["generic"] }` will not be sent despite matching the first, less specific filter because it did not match the second filter which was the most relevant and specific
```

Comment:
```suggestion
 * @example
 * With the following configuration, an event `{ namespace: "A.B.C", categories: ["generic"] }` will not be sent despite matching the first, less specific filter because it did not match the second filter which was the most relevant and specific
```

Reply:
Updated.